crates/core/src/client_events/mod.rs

```diff
@@ -356,10 +360,69 @@ enum Error {
 }
 
 #[inline]
+async fn handle_proximity_cache_info_query(
+    proximity_cache: &Arc<crate::node::proximity_cache::ProximityCacheManager>,
+) -> freenet_stdlib::client_api::ProximityCacheInfo {
+    let (my_cache_hashes, neighbor_cache_data) = proximity_cache.get_introspection_data().await;
+    let stats = proximity_cache.get_stats().await;
+
+    let my_cache = my_cache_hashes
+        .into_iter()
+        .map(|hash| freenet_stdlib::client_api::ContractCacheEntry {
+            contract_key: format!("hash_{:08x}", hash),
```

Comment:
@sanity @iduartgomez Considering that the cache hash is only generated from the first 4 bytes and is prefixed, couldn't this contaminate the traceability?

Reply:
@netsirius Correct - the `hash_{:08x}` format makes it impossible to trace back to actual contracts. This introspection API is essentially useless for debugging without storing the full ContractKey alongside the hash.

Reply:
@netsirius Correct - this is now fixed. The introspection API now returns the actual `ContractKey` string (via `contract_key.to_string()`) instead of the `hash_{:08x}` format, making it fully traceable for debugging.